Modules/Loadable/Markups/MRML/vtkMRMLMarkupsCurveNode.cxx

```diff
@@ -183,6 +194,182 @@ double vtkMRMLMarkupsCurveNode::GetCurveLengthBetweenStartEndPointsWorld(vtkIdTy
   }
 }
 
+//---------------------------------------------------------------------------
+void vtkMRMLMarkupsCurveNode::ResampleCurveSurface(double controlPointDistance, vtkMRMLModelNode* modelNode, double maximumSearchRadius)
+{
+	if(!modelNode)
+	  {
+	  vtkErrorMacro("vtkMRMLMarkupsCurveNode::ResampleCurveSurface failed: Constraint surface is not valid");
+	  }
+	vtkPoints* originalPoints = this->GetCurvePointsWorld();
+	if (!originalPoints  || originalPoints->GetNumberOfPoints() < 2)
+      {
+	  return;
+      }
+	vtkNew<vtkPoints> originalControlPoints;
+	this->GetControlPointPositionsWorld(originalControlPoints);
+	vtkNew<vtkStringArray> originalLabels;
+	this->GetControlPointLabels(originalLabels);
+
+	vtkNew<vtkPoints> interpolatedPoints;
+	vtkMRMLMarkupsCurveNode::ResamplePoints(originalPoints, interpolatedPoints, controlPointDistance, this->CurveClosed);
+
+	vtkPolyData* surfacePolydata = modelNode->GetPolyData();
```

Comment:
You need to transform the model polydata to the world coordinate system (if parent transform node is set then get TransformToWorld transform and use it in vtkTransformFilter)